scripts/pkgconfig-export-targets.cmake.in

```diff
@@ -9,7 +9,7 @@ include(hunter_pkgconfig_export_target)
 
 hunter_pkgconfig_export_target("@PKG_CONFIG_MODULE@" "@PKG_GENERATE_SHARED@")
 
-foreach(_hunter_deps @DEPENDS_ON_PACKAGES@)
+foreach(_hunter_deps @DEPENDS_ON_PKGCONFIGS@)
```

Comment:
Yeah, I find that `set(DEPENDS_ON_PACKAGES)` line a little suspect as well.  I think the "pkgconfig-export-targets.cmake.in" change is correct.  It doesn't make sense to add `PkgConfig::${dep}` dependencies for things in `DEPENDS_ON_PACKAGES`, since they aren't necessarily in the `PkgConfig` namespace!  Whereas the things listed on `DEPENDS_ON_PKGCONFIGS` would be in the `PkgConfig` namespace.

Some packages may need to be updated if the `DEPENDS_ON_PACKAGES`/`DEPENDS_ON_PKGCONFIG` values are swapped around.  But, there don't seem to be many autotools-based packages that actually use these options.

Reply:
I was trying to hunterize libarchive long ago (June 2020 😄 ). I was using the method of forking the cmake and adding hunter magic. Here is how far I got https://github.com/hjmallon/libarchive/commit/d31252161478985aa7790d72a25734fa65062424

I have no idea whether that will be any help as a starting point for avoiding autotools here, but I assume that would avoid this issue?

Reply:
> I have no idea whether that will be any help as a starting point for avoiding autotools here, but I assume that would avoid this issue?

I don't see a reason to avoid autotools and even if we did for libarchive, I still think the behavior of `DEPENDS_ON_PACKAGES` is wrong.  And, as I mentioned, there aren't many places where DEPENDS_ON_PACKAGES is used with autotools.

Reply:
Okay, I will merge this based on your testing. If autotools packages are failing we may have to revisit this issue.